src/PreserveModuleNamePlugin.ts

```diff
@@ -38,15 +135,8 @@ export class PreserveModuleNamePlugin {
           if (module.constructor.name === "ConcatenatedModule")
             realModule = module["rootModule"];
 
-          let preserve: string | true | undefined = realModule[preserveModuleName];
-          let id = typeof preserve === "string" ? preserve : null;
-
-          // No absolute request to preserve, we try to normalize the module resource
-          if (!id && realModule.resource)
-            id = fixNodeModule(realModule, modulesBeforeConcat) || 
-                 makeModuleRelative(roots, realModule.resource) ||
-                 aliasRelative(alias, realModule.resource);
-                    
+          // Get the module id
+          let id = getModuleId(realModule, roots, alias);
```

Comment:
If we factor out `getModuleId`, we might as well take the `if (!id) throw` below with it.
That changes the return type to a non-null value.

Reply:
What do you mean by this?